File: backend/app/models/eligibility.py
from datetime import date, datetime
from enum import StrEnum

from pydantic import BaseModel


class CoverageStatus(StrEnum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    FAILED = "failed"
    PENDING = "pending"


class EligibilityCheckBase(BaseModel):
    client_id: str
    payer: str
    check_date: datetime
    coverage_status: CoverageStatus
    effective_date: date | None = None
    termination_date: date | None = None
    plan_name: str | None = None
    failure_reason: str | None = None
    source: str = "manual"
    processed: bool = True

File: frontend/lib/validation/eligibility.ts
import { z } from "zod";

export const eligibilityFormSchema = z.object({
  client_id: z.string().min(1, "Select a client"),
  payer: z.string().min(1, "Required"),
  coverage_status: z.enum(["active", "inactive", "failed", "pending"]),
  plan_name: z.string().optional(),
  failure_reason: z.string().optional(),
});

export type EligibilityFormValues = z.infer<typeof eligibilityFormSchema>;
